tools/cli/prepNextRelease.js

```diff
@@ -1,56 +1,122 @@
+  if (diff || diffStaged) {
+    throw 'You have uncommitted changes. Please commit your changes before proceeding.\nAborting!';
+  }
+  const gitBranch = await execShellCommand('git branch');
+  const branches = gitBranch.split('\n');
+  const currentBranch = branches.find(branch => branch.startsWith('*'));
+  console.log(currentBranch);
+  if (currentBranch !== '2.0-preview') {
+    process.stdout.write('You are not on 2.0-preview branch. Do you want to checkout? (y/n) ');
+    if (!(await boolPrompt())) {
+      throw 'User aborted!';
+    }
+    await execShellCommand('git checkout 2.0-preview');
+  }
+  await execShellCommand('git pull');
+};
 
-// append the suffix to form a new version
-let nextVersion = version + '-' + versionSuffix;
-console.log('@next version: ' + nextVersion);
+const updatePackageVersion = async () => {
+  const relativePathToTeamsjsPackageJson = '../../packages/teams-js/package.json';
+  const newVersion = getNewPackageVersion();
+  console.log(`Updating package.json to ${newVersion}`);
+  const absolutePathToTeamsjsPackageJson = path.resolve(__dirname, relativePathToTeamsjsPackageJson);
+  if (!fs.existsSync(absolutePathToTeamsjsPackageJson)) {
+    throw `ERROR: ${absolutePathToTeamsjsPackageJson} was not found.`;
+  }
+  const packageJson = fs.readFileSync(absolutePathToTeamsjsPackageJson, 'utf8');
+  const newPackageJson = packageJson.replace(`"version": "${currentPackageVersion}"`, `"version": "${newVersion}"`);
+  fs.writeFileSync(absolutePathToTeamsjsPackageJson, newPackageJson);
+};
 
-// update package.json with the new version
-packageJson.version = nextVersion;
+const buildAndUpdateIntegrityHash = async () => {
+  console.log('Updating integrity hash');
+  await execShellCommand('cd packages/teams-js & yarn build & node prepNewReadme.js');
+};
 
-// Sets result in Azure devops pipeline output variable 'NextVersion'
-console.log(`##vso[task.setvariable variable=NextVersion;isOutput=true]${nextVersion}`);
+const checkoutAndPushReleaseBranch = async () => {
+  console.log('Please validate all uncommitted changes: \n');
+  const diff = await execShellCommand('git diff -U0');
+  console.log(diff);
+  process.stdout.write('Proceed? (y/n) ');
+  if (!(await boolPrompt())) {
+    throw 'User aborted!';
+  }
+  const newVersion = getNewPackageVersion();
+  const branchName = `release/${newVersion}`;
+  await execShellCommand(`git checkout -b ${branchName}`);
+  await execShellCommand('git add -A');
+  await execShellCommand(`git commit -m "Releasing ${newVersion}"`);
+  await execShellCommand(`git push --set-upstream origin ${branchName}`);
+};
 
-savePackageJson(packageJson);
+(async () => {
+  try {
+    await checkoutAndPullMainBranch();
+    await updatePackageVersion();
+    await buildAndUpdateIntegrityHash();
+    await checkoutAndPushReleaseBranch();
+    // TODO: Add code to create the PR automatically using github api
+  } catch (e) {
+    console.log(e);
+  }
```

Comment:
Might be good to have a step that prints something like "Hey FYI this is going to actually release, are you sure? (Y/N)"

Reply:
There's a prompt in the checkoutAndPushRelease function to make sure all the changes are valid before commit/push. I'll add one more at the start of the process.

Reply:
I added a prompt at the start of the script. So whenever this is run, you'll get: 
- one at the start to validate you want to release the right version
- one to checkout 2.0-preview if you're not on it (so that you don't lose any code)
- one to check the changes in package.json and readme.md (and potentially other files that need to be changed)